crates/common/src/zk_compile.rs

```diff
@@ -277,14 +280,31 @@ impl ZkSolc {
         let mut displayed_warnings = HashSet::new();
         let mut data = BTreeMap::new();
         // Step 1: Collect Source Files
-        let sources = self.get_versioned_sources().wrap_err("Cannot get source files")?;
+        let sources = match self.get_versioned_sources() {
+            Ok(sources) => sources,
+            Err(e) => return Err(eyre::eyre!("Could not resolve sources: {}", e)),
+        };
         let mut contract_bytecodes = BTreeMap::new();
 
         // Step 2: Compile Contracts for Each Source
         for (_solc, version) in sources {
             info!("\nCompiling {} files...", version.1.len());
             //configure project solc for each solc version
             for (contract_path, _) in version.1 {
+                // Check if the contract_path is in 'sources' directory or its subdirectories
+                let is_in_sources_dir = contract_path
+                    .ancestors()
+                    .any(|ancestor| ancestor.starts_with(&self.project.paths.sources));
+
+                let is_in_scripts_dir = contract_path
+                    .ancestors()
+                    .any(|ancestor| ancestor.starts_with(&self.project.paths.scripts));
+
+                // Skip this file if it's not in the 'sources' directory or its subdirectories
+                if !is_in_sources_dir && !is_in_scripts_dir {
+                    continue
+                }
+
```

Comment:
This shouldnt be needed since: https://github.com/matter-labs/foundry-zksync/pull/210

Reply:
Thought so also but wasn't sure if Nick had this for another reason - I saw that all sources came from these folders anyways